xformers/components/attention/core.py

```diff
@@ -219,20 +223,31 @@ def scaled_dot_product_attention(
         or (att_mask is not None and att_mask.is_sparse)
     )
 
-    if att_mask is not None and q.shape[-2] < att_mask.shape[0]:
-        # The sequence is smaller than the mask
+    # Try to handle a case where the sequence is smaller than the mask
+    if (
+        att_mask is not None
+        and q.shape[-2] == k.shape[-2]
+        and q.shape[-2] < att_mask.shape[0]
+    ):
         seq = q.shape[-2]
-
-        if not att_mask.is_sparse:
-            att_mask = att_mask[:seq, :seq]
```

Comment:
we were testing for smaller sequences, but when the mask was sparse it was not adjusted -> possible memory error which showed up on CI later on